pkg/urbit/vere/io/ames.c

```diff
@@ -771,29 +828,44 @@ _ames_recv_cb(uv_udp_t*        wax_u,
   {
     pas_o = c3n;
 
-    //  if the queue is full, and we can't forward synchronously,
+    u3_weak lac;
+    //  if the recipient is a galaxy, their lane is always &+~gax
+    //
+    if ( (rec_d[1] == 0) && (256 > rec_d[0]) ) {
+      lac = u3nc(c3y, (c3_y)rec_d[0]);
+    }
+    //  otherwise, try to get the lane from cache
+    //
+    else {
+      lac = _ames_lane_from_cache(sam_u->lax_p, u3i_chubs(2, rec_d));
+    }
+
+    //  if we don't know the lane, and the scry queue is full,
     //  just drop the packet
     //
     //TODO  drop oldest item in forward queue in favor of this one.
     //      ames.c doesn't/shouldn't know about the shape of scry events,
     //      so can't pluck these out of the event queue like it does in
     //      _ames_cap_queue. as such, blocked on u3_lord_peek_cancel or w/e.
     //
-    if ( (1000 < sam_u->foq_d)
-      && !(rec_d[1] == 0 && (256 > rec_d[0])) )
+    if ( (u3_none == lac) && (1000 < sam_u->foq_d) )
     {
       c3_free(con_y);
 
       sam_u->fod_d++;
-      if ( 0 == (sam_u->fod_d % 100000) ) {
+      if ( 0 == (sam_u->fod_d % 10000) ) {
         u3l_log("ames: dropped %" PRIu64 " forwards total\n", sam_u->fod_d);
       }
     }
+    //  if we know there's no lane, drop the packet
+    //
+    else if (u3_nul == lac) {
```

Comment:
There's some inconsistent conditional styling in this file now. House style is to have whitespace around the conditional expression (ie, `( u3_nul == lac )`).